File: lib/data/models/foods/food_search_model.dart
import 'package:calory_tool/data/models/foods/food_model.dart';
import 'package:hive_flutter/hive_flutter.dart';

part 'food_search_model.g.dart';

@HiveType(typeId: 7)
class FoodSearchModel extends HiveObject {
  FoodSearchModel({required this.foods, this.pageNumber});

  factory FoodSearchModel.fromJson(Map<String, dynamic> json) {
    final foods = json['foods'] as List? ?? [];
    return FoodSearchModel(
      pageNumber: (json['page_number'] as num?)?.toInt(),
      foods:
          foods
              .map((dynamic e) => FoodModel.fromJson(e as Map<String, dynamic>))
              .toList(),
    );
  }

  @HiveField(0)
  final int? pageNumber;
  @HiveField(1)
  final List<FoodModel> foods;
}

File: server/src/models/foods/FoodSearchModel.js
import FoodModel from "./FoodModel.js";

class FoodSearchModel {
    constructor({ page_number, foods }) {
        this.page_number = page_number;
        this.foods = foods;
    }

    static fromJson(json) {
        const foodsJson = json["foods"] || {};
        const foodArray = foodsJson["food"] || null;
        const pageNumber = foodsJson["page_number"] ? parseInt(foodsJson["page_number"]) : null;
        const foods = foodArray ? foodArray.map(food => FoodModel.fromJson(food)) : null;

        return new FoodSearchModel({
            page_number: pageNumber,
            foods: foods
        });
    }
}

export default FoodSearchModel;